app/uk/gov/hmrc/gform/gform/SummaryController.scala

```diff
@@ -216,12 +221,22 @@ class SummaryController(
       maybeAccessCode,
       OperationWithForm.DownloadSummaryPdf
     ) { implicit request => implicit l => cache => implicit sse => formModelOptics =>
-      pdfService
-        .generateSummaryPDF(maybeAccessCode, cache, SummaryPagePurpose.ForUser, formModelOptics)
-        .map { pdfStream =>
+      pdfRenderService
+        .createPDFHtml[DataOrigin.Mongo, SectionSelectorType.Normal, PDFType.Summary](
+          request.messages.messages(
+            "summary.checkYourAnswers"
+          ) + " - " + cache.formTemplate.formName.value + " - GOV.UK",
+          Option(cache.formTemplate.summarySection.title.value()),
+          cache,
+          formModelOptics,
+          Option(PDFModel.HeaderFooter(Option(cache.formTemplate.summarySection.header), None)),
+          None
+        )
+        .flatMap(pdfGeneratorService.generatePDFLocal)
+        .map { pdfBytes =>
           Result(
             header = ResponseHeader(200, Map.empty),
-            body = HttpEntity.Streamed(pdfStream, None, Some("application/pdf"))
+            body = HttpEntity.Strict(ByteString(pdfBytes), Some("application/pdf"))
```

Comment:
Just a question, why streaming abandoned?

Reply:
Was struggling a bit to get Stream working with java.io. API, I think I have a working piece of code now.